src/providers/interface-provider.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Provider interface file
+ *
+ * @package wp-page-cache-control
+ */
+
+namespace Alley\WP\WP_Page_Cache_Control\Providers;
+
+use WP_Post;
+use WP_Term;
+
+/**
+ * Base provider contract.
+ *
+ * @todo Allow URLs to be registered that will be purged when a post/term is purged.
+ * @todo Add method to perform the purge of the queued URLs.
+ */
+interface Provider {
+	/**
+	 * Set the TTL for the cache for the current request.
+	 *
+	 * @param int $seconds TTL in seconds.
+	 * @return void
+	 */
+	public function ttl( int $seconds ): void;
+
+	/**
+	 * Set the default TTL for the cache for all REST API requests.
+	 *
+	 * @param int $seconds TTL in seconds.
+	 * @return void
+	 */
+	public function ttl_rest_api( int $seconds ): void;
+
+	/**
+	 * Disable the page cache for the current request.
+	 *
+	 * @return void
+	 */
+	public function disable_cache(): void;
+
+	/**
+	 * Disable the page cache for the user for this and all subsequent requests.
+	 *
+	 * @return void
+	 */
+	public function disable_cache_for_user(): void;
+
+	/**
+	 * Enable the page cache for the user for this and all subsequent requests.
+	 *
+	 * @return void
+	 */
+	public function enable_cache_for_user(): void;
+
+	/**
+	 * Check if the page cache is disabled for the current request.
+	 *
+	 * @return bool
+	 */
+	public function is_user_cache_disabled(): bool;
+
+	/**
+	 * Register a cache group.
+	 *
+	 * @param array<int, string> $groups The groups to register.
+	 * @return void
+	 */
+	public function register_groups( array $groups ): void;
+
+	/**
+	 * Register a cache group.
+	 *
+	 * @param string $group The group to register.
+	 * @return void
+	 */
+	public function register_group( string $group ): void;
+
+	/**
+	 * Retrieve the registered cache groups.
+	 *
+	 * @return array<int, string> The registered groups.
+	 */
+	public function groups(): array;
```

Comment:
Would this be clearer as `get_groups()`?

Reply:
Agree!